src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -72,6 +72,15 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedValueException($value, 'string');
         }
 
+        if (!$constraint->allowSpaces && preg_match('/\s/', $value)) {
```

Comment:
Using `\s` contradicts with replacing only spaces just below. The check should be `false !== strpos($value, ' ')`